plugins/modules/general_repository.py

```diff
@@ -410,6 +355,12 @@ def run_module():
                             hooks_url = "%s/hooks" % (url)
                         full_name = "%s/%s" % (
                             module.params['organization'], module.params['repository'])
+
+                        is_private = module.params['visibility']
+
+                        if module.params['visibility'] == 'internal' or module.params['visibility'] == 'private':
+                            is_private = True
```

Comment:
Would this also need an else statement to make it false? Otherwise it would be set to 'public'